File: backend/core/utils/debug.py
from datetime import datetime
from django.conf import settings


def debug_print(message: str, prefix: str = "SYSTEM") -> None:
    """
    Advanced, styled diagnostic printer for development.
    Features: ANSI colors, timestamps, and production safety.
    """
    if settings.DEBUG:
        # High-contrast color palette
        color_map = {
            "GO-AUTH": "\033[95m",  # Magenta
            "RISK": "\033[91m",  # Red
            "ENRICH": "\033[93m",  # Yellow
            "SUCCESS": "\033[92m",  # Green
            "FALLBACK": "\033[33m",  # Brown/Orange
            "SYSTEM": "\033[96m",  # Cyan
            "CELERY": "\033[94m",  # Blue
        }

        # Pick color based on prefix or default to Cyan
        color = color_map.get(prefix.upper(), "\033[96m")
        bold = "\033[1m"
        gray = "\033[90m"
        reset = "\033[0m"

        timestamp = datetime.now().strftime("%H:%M:%S.%f")[:-3]
        icon = "🔌"  # High-speed connectivity icon

        # Using fixed-width padding for the prefix to ensure vertical alignment
        output = (
            f"{gray}[{timestamp}]{reset} "
            f"{color}{bold}{icon} {prefix:<10}:{reset} "
            f"{message}"
        )
        print(output, flush=True)

File: services/go/shared/platform/debug/debug.go
package debug

import (
	"fmt"
	"strings"
	"time"
)

const (
	colorMagenta = "\033[95m"
	colorRed     = "\033[91m"
	colorYellow  = "\033[93m"
	colorGreen   = "\033[92m"
	colorOrange  = "\033[33m"
	colorCyan    = "\033[96m"
	colorBlue    = "\033[94m"
	colorGray    = "\033[90m"
	bold         = "\033[1m"
	reset        = "\033[0m"
)

var colorMap = map[string]string{
	"GO-AUTH": colorMagenta,
	"RISK":    colorRed,
	"ENRICH":  colorYellow,
	"SUCCESS": colorGreen,
	"FALLBACK": colorOrange,
	"SYSTEM":  colorCyan,
	"CELERY":  colorBlue,
	"GO-CHAT": colorCyan,
}

// Print diagnostic messages with ANSI colors and timestamps, matching the Python debug_print style.
func Print(prefix, message string) {
	color, ok := colorMap[strings.ToUpper(prefix)]
	if !ok {
		color = colorCyan
	}

	timestamp := time.Now().Format("15:04:05.000")
	icon := "🔌"

	fmt.Printf("%s[%s]%s %s%s%s %-10s:%s %s\n",
		colorGray, timestamp, reset,
		color, bold, icon, prefix, reset,
		message,
	)
}
